# HG changeset patch
# User Ben Abbott  <bpabbott@mac.com>
# Date 1314829157 14400
#      Wed Aug 31 18:19:17 2011 -0400
# Branch stable
# Node ID d2997525fcb6343609cfca2e1df9ac0da6cbd53e
# Parent  9b191d0e547f9a760d9d046b7ae16cf1ddc3877d
Fix for bug #34158.

* plot/axis.m: Check that lower axis limits are less han upper limits.

diff --git a/scripts/plot/axis.m b/scripts/plot/axis.m
--- a/scripts/plot/axis.m
+++ b/scripts/plot/axis.m
@@ -272,18 +272,18 @@ function curr_axis = __axis__ (ca, ax, v
 
     len = length (ax);
 
     if (len != 2 && len != 4 && len != 6)
       error ("axis: expecting vector with 2, 4, or 6 elements");
     endif
 
     for i = 1:2:len
-      if (ax(i) == ax(i+1))
-        error ("axis: limits(%d) cannot equal limits(%d)", i, i+1);
+      if (ax(i) >= ax(i+1))
+        error ("axis: limits(%d) must be less than limits(%d)", i, i+1);
       endif
     endfor
 
     if (len > 1)
       set (ca, "xlim", [ax(1), ax(2)]);
     endif
 
     if (len > 3)
